### DISCUSSION TITLE
Prevent deletion of edges during  node deletion

### DISCUSSION THREAD

**Discussion by MSB89** [bcakmakoglu/vue-flow#1605] [Q&A] [answered]
I want to prevent the deletion of specific nodes, I can do this in ```onNodesChange``` with a check on ```change.type === 'remove'``` and then apply my own logic to determine if the node should or shouldn't be deleted. 

The problem I'm having is that ```onEdgesChange``` is firing before ```onNodesChange```. I want to prevent the edges from being deleted if they are part of the node deletion process. I can't simply check for ```change.type === 'remove'``` etc because I want to be able to individually delete the edges, I only want to prevent the deletion during the node deletion process - if that node can't be deleted. I can't seem to work round the ```onEdgesChange``` firing first.

**Comment by bcakmakoglu** [comment, marked as answer]
the `removeNodes` action allows you to specify that "orphaned" edges should not be removed (it will cause a warning though because edges need a source and a target node).

```ts
removeNodes(nodesOrNode, false)
```
